components/org.wso2.carbon.secvault.securevault/src/main/java/org/wso2/carbon/secvault/securevault/cipher/JKSBasedCipherProvider.java

```diff
@@ -89,14 +90,20 @@ public void init(SecretRepositoryConfiguration secretRepositoryConfiguration, Li
     }
 
     private KeyStore loadKeyStore(String keyStorePath, char[] keyStorePassword) throws SecureVaultException {
-            Path keyStoreFileLocation;
-            if (SecureVaultUtils.isOSGIEnv()) {
-                keyStoreFileLocation = Paths.get(SecureVaultUtils.getCarbonHome().get().toString(), keyStorePath);
-            } else {
-                keyStoreFileLocation = Paths.get(keyStorePath);
+        Path keyStoreFileLocation;
+        // TODO: Get keystore path with relevant to the runtime
+        if (SecureVaultUtils.isOSGIEnv()) {
```

Comment:
Why do we check this everywhere? Can't we handle this all at one level, i.e., one place where we have OSGi and non-OSGi code and everywhere else it is common. AFAIR according to the code review it was suggested to have the OSGi and non-OSGi check in the component at one place.